extensions/ql-vscode/src/query-results.ts

```diff
@@ -49,45 +49,32 @@ export interface InitialQueryInfo {
 }
 
 export class CompletedQueryInfo implements QueryWithResults {
-  readonly query: QueryEvaluationInfo;
-  readonly message?: string;
-  readonly successful?: boolean;
-  /**
-   * The legacy result. This is only set when loading from the query history.
-   */
-  readonly result: legacyMessages.EvaluationResult;
-  readonly logFileLocation?: string;
-  resultCount: number;
-
-  /**
-   * Map from result set name to SortedResultSetInfo.
-   */
-  sortedResultsInfo: Record<string, SortedResultSetInfo>;
-
-  /**
-   * How we're currently sorting alerts. This is not mere interface
-   * state due to truncation; on re-sort, we want to read in the file
-   * again, sort it, and only ship off a reasonable number of results
-   * to the webview. Undefined means to use whatever order is in the
-   * sarif file.
-   */
-  interpretedResultsSortState: InterpretedResultsSortState | undefined;
-
-  /**
-   * Note that in the {@link readQueryHistoryFromFile} method, we create a CompletedQueryInfo instance
-   * by explicitly setting the prototype in order to avoid calling this constructor.
-   */
-  constructor(evaluation: QueryWithResults) {
-    this.query = evaluation.query;
-    this.logFileLocation = evaluation.logFileLocation;
-    this.result = evaluation.result;
-
-    this.message = evaluation.message;
-    this.successful = evaluation.successful;
-
-    this.sortedResultsInfo = {};
-    this.resultCount = 0;
-  }
+  constructor(
+    public readonly query: QueryEvaluationInfo,
+
+    /**
+     * The legacy result. This is only set when loading from the query history.
+     */
+    public readonly result: legacyMessages.EvaluationResult,
+    public readonly logFileLocation?: string,
```

Comment:
Should this be 
```ts
public readonly logFileLocation: string | undefined,
```

Like we do in `LocalQueryInfo`?

Same question for other optional members.

Reply:
Hm, in this case the original used `?` already, so I would rather keep it as is for now.